server/app/durablejobs/jobs/CalculateEligibilityDeterminationJob.java

```diff
@@ -0,0 +1,77 @@
+package durablejobs.jobs;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import durablejobs.DurableJob;
+import io.ebean.DB;
+import io.ebean.Database;
+import io.ebean.Transaction;
+import javax.inject.Inject;
+import models.ApplicationModel;
+import models.EligibilityDetermination;
+import models.PersistedDurableJobModel;
+import models.ProgramModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import services.applicant.ApplicantService;
+import services.applicant.ReadOnlyApplicantProgramService;
+import services.program.ProgramDefinition;
+
+/**
+ * Calculate eligibility determination for application submitted before pre-compute eligibility
+ * feature was implemented.
+ */
+public final class CalculateEligibilityDeterminationJob extends DurableJob {
+  private static final Logger logger =
+      LoggerFactory.getLogger(CalculateEligibilityDeterminationJob.class);
+  private final ApplicantService applicantService;
+
+  private final Database database;
+  private final PersistedDurableJobModel persistedDurableJobModel;
+
+  @Inject
+  public CalculateEligibilityDeterminationJob(
+      ApplicantService applicantService, PersistedDurableJobModel persistedDurableJobModel) {
+    this.applicantService = checkNotNull(applicantService);
+    this.persistedDurableJobModel = checkNotNull(persistedDurableJobModel);
+    this.database = DB.getDefault();
+  }
+
+  @Override
+  public PersistedDurableJobModel getPersistedDurableJob() {
+    return persistedDurableJobModel;
+  }
+
+  @Override
+  public void run() {
+    logger.info("Starting job to calculate eligibility determination.");
+
+    try (Transaction jobTransaction = database.beginTransaction()) {
+      String filter =
+          """
+          eligibility_determination = 'NOT_COMPUTED'
+          """;
+      try (var query = database.find(ApplicationModel.class).where().raw(filter).findIterate()) {
```

Comment:
(blocking) This should also filter for applications that have `LifecycleStage.Active` or `LifecycleStage.Obsolete`. That should cut down on a lot of empty draft applications.

It would be nice to not have to do obsoleted records as they are superseded by the active application, but they are returned in the API. Unless the PRD/TDD indicates we can skip them best to include them.

Reply:
+1

Reply:
(also not blocking at all but it should be possible to build the filter in the ebeans syntax instead of as a raw strong)